src/content/en/updates/2018/03/smooshgate.md

```diff
@@ -0,0 +1,217 @@
+project_path: /web/_project.yaml
+book_path: /web/updates/_book.yaml
+description: Recently the internet was on fire because of #SmooshGate. What is it about, and where do we go from here? This write-up gives an overview.
+
+{# wf_updated_on: 2018-03-19 #}
+{# wf_published_on: 2018-03-19 #}
+{# wf_tags: javascript #}
+{# wf_featured_image: /web/updates/images/generic/info.png #}
+{# wf_featured_snippet: Recently the internet was on fire because of #SmooshGate. What is it about, and where do we go from here? This write-up gives an overview. #}
+{# wf_blink_components: Blink>JavaScript>Language #}
+
+# #SmooshGate FAQ {: .page-title }
+
+{% include "web/_shared/contributors/mathiasbynens.html" %}
+
+## What the _smoosh_ happened?! {: #what }
+
+A [proposal](https://tc39.github.io/proposal-flatMap/) for a JavaScript
+language feature called `Array.prototype.flatten` turns out to be
+Web-incompatible. Shipping the feature in Firefox Nightly caused [at least
+one popular website](https://bugzilla.mozilla.org/show_bug.cgi?id=1443630)
+to break. Given that the problematic code is part of the (once extremely
+popular) MooTools library, it’s likely that many more websites are affected.
+
+The proposal author jokingly suggested renaming `flatten` to `smoosh` to
+avoid the compatibility issue. The joke was not clear to everyone, some
+people started to incorrectly believe that the new name had already been
+decided, and things escalated quickly.
+
+## What does `Array.prototype.flatten` do? {: #flatten }
+
+`Array.prototype.flatten` flattens arrays recursively up to the specified
+`depth`, which defaults to `1`.
+
+```js
+// Flatten one level:
+const array = [1, [2, [3]]];
+array.flatten();
+// → [1, 2, [3]]
+
+// Flatten recursively until the array contains no more nested arrays:
+array.flatten(Infinity);
+// → [1, 2, 3]
+```
+
+The same proposal includes `Array.prototype.flatMap`, which is like
+`Array.prototype.map` except it flattens the result into a new array.
+
+```js
+[2, 3, 4].flatMap((x) => [x, x * 2]);
+// → [2, 4, 3, 6, 4, 8]
+```
+
+## What is MooTools doing wrong? {: #mootools }
```

Comment:
Was Mootools wrong at the time? It reads like an accusation, when this could have easily been de rigueur at the time of it's creation.... It's only in hindsight we learn it was bad practice.

Reply:
Clarified this by rephrasing the section. I think the title is fair as it was a common question during #SmooshGate.

Reply:
sg.